R/calcHeatingSystem.R

```diff
@@ -21,14 +22,22 @@ calcHeatingSystem <- function(subtype = c("Purchasing cost", "Efficiency")) {
 
   subtype <- match.arg(subtype)
 
+  # all heating technologies
+  hs <- getBrickMapping("heatingSystem.csv")
+
   # map heating technologies
   hsMap <- toolGetMapping("technologyMapping_EU_ReferenceScenario.csv",
                           type = "sectoral", where = "mredgebuildings",
                           returnPathOnly = TRUE) %>%
     read.csv(comment.char = "") %>%
     select(-"comment") %>%
     pivot_longer(matches("^weight"), names_to = "typ", values_to = "weight") %>%
-    mutate(typ = sub("^weight(.*)$", "\\1", .data[["typ"]]))
+    mutate(typ = sub("^weight(.*)$", "\\1", .data[["typ"]])) %>%
+    right_join(unique(hs["hs"]), by = c(technologyBRICK = "hs"))
+
+  if (any(is.na(hsMap))) {
+    stop("Incomplete mapping of heating technologies.")
+  }
```

Comment:
I actually want this to be strict and to not allow any NAs. I'd rather take the risk of creating false positives here since I don't expect a NA in any column. I'd rather want this code to fail then have missing entries in BRICK.